### DISCUSSION TITLE
Validate based on input value and other fields value (Inspector)

### DISCUSSION THREAD

**Discussion by pandika-csgi** [clientIO/joint#2264] [Q&A] [answered]
### Introduction

Hi, how can I validate input field on inspector based on value itself and other fields value?

e.g. 
https://resources.jointjs.com/docs/rappid/v3.7/ui.Inspector.html#ui.Inspector.validation
```js
validateProperty: function(path, value) {
  switch (path) {
      case 'attrs/body/stroke':
      case 'attrs/body/fill':
          if (this.REGEX_HEXCOLOR.test(value)) break;
          return 'Invalid Color (e.g. #ff0000)';
      case 'attrs/body/strokeWidth':
          if (_.isNumber(value) && value >= 0) break;
          return 'Invalid Stroke Width (A positive number)';
      case 'phoneNumber':
          if (this.REGEX_PHONE_NUMBER.test(value)) break;
          return 'Invalid Phone Number (e.g. 123-456-7890)';
      case 'emailAddress':
          if (this.REGEX_EMAIL_ADDRESS.test(value)) break;
          return 'Invalid Email Address.';
  }
  return null;
 },
```

in `validateProperty` method let's say I want to check `phoneNumber` regex if emailAddress is already filled or validate`strokeWidth` if `phoneNumber` is valid and `emailAddress` is valid

how to get other values in validateProperty?

Thank you

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment, marked as answer]
`validateProperty()` is a method of the cell. It means you can read any value from the model.
```js
const strokeWidth = this.prop('attrs/body/strokeWidth');
const phoneNumber = this.prop('phoneNumber');
```
